apps/dashboard/app/(app)/settings/root-keys/new/client.tsx

```diff
@@ -91,9 +91,65 @@ export const Client: React.FC<Props> = ({ apis }) => {
         </CardHeader>
         <CardContent>
           <div className="flex flex-col gap-4">
+            <PermissionToggle
+              key="selectAll-all"
+              permissionName="selectAll-all"
+              label="Select All Permissions"
+              description="Grants all permissions across for this workspace."
+              checked={Object.values(workspacePermissions)
+                .flatMap((category) => Object.values(category).map(({ permission }) => permission))
+                .every((permission) => selectedPermissions.includes(permission))}
+              setChecked={(c) => {
+                const allPermissions = Object.values(workspacePermissions).flatMap((category) =>
+                  Object.values(category).map(({ permission }) => permission),
+                );
+
+                if (c) {
+                  allPermissions.forEach((permission) => {
+                    if (!selectedPermissions.includes(permission)) {
+                      handleSetChecked(permission, true);
+                    }
+                  });
+                } else {
+                  allPermissions.forEach((permission) => {
+                    if (selectedPermissions.includes(permission)) {
+                      handleSetChecked(permission, false);
+                    }
+                  });
+                }
+              }}
+            />
             {Object.entries(workspacePermissions).map(([category, allPermissions]) => (
               <div key={`workspace-${category}`} className="flex flex-col gap-2">
-                <span className="font-medium">{category}</span>{" "}
+                <span className="font-medium">{category}</span>
+                <PermissionToggle
+                  key={`selectAll-${category}`}
+                  permissionName={`selectAll-${category}`}
+                  label={`Select All under ${category}`}
+                  description={`Grants all permissions under ${category}.`}
+                  checked={Object.values(allPermissions).every(({ permission }) =>
+                    selectedPermissions.includes(permission),
+                  )}
+                  setChecked={(c) => {
+                    const categoryPermissions = Object.values(allPermissions).map(
+                      ({ permission }) => permission,
+                    );
+
+                    if (c) {
+                      categoryPermissions.forEach((permission) => {
+                        if (!selectedPermissions.includes(permission)) {
+                          handleSetChecked(permission, true);
+                        }
+                      });
+                    } else {
+                      categoryPermissions.forEach((permission) => {
+                        if (selectedPermissions.includes(permission)) {
+                          handleSetChecked(permission, false);
+                        }
+                      });
+                    }
+                  }}
+                />
```

Comment:
Let's move this to a separate function and avoid using abbreviations like `c` and maybe use `isChecked` instead.

And, I believe we can make this a bit shorter like this.

```
setChecked={(c) => {
  const categoryPermissions = Object.values(allPermissions).map(({ permission }) => permission);
  categoryPermissions.forEach(permission => {
    if (selectedPermissions.includes(permission) !== c) {
      handleSetChecked(permission, c);
    }
  });
}}
```

Reply:
I agree. Let's not use something like C. But you u cannot really make it shorter. Running ```pnpm fmt``` is going to put spaces and make it a little longer than usual.

Reply:
Did you see the code I sent you above?